src/compile/data/dataflow.ts

```diff
@@ -127,3 +135,23 @@ export interface DependentNode {
    */
   dependentFields: () => StringSet;
```

Comment:
Can't we just make produces return `null` for nodes that don't depend on others?